src/viking.js

```diff
@@ -1,11 +1,206 @@
+
+  receiveDamage (damage) {
+    this.health -= damage;
+
+    if (this.health > 0) {
+      return `A Saxon has received ${damage} points of damage`;
+    } else {
+      return `A Saxon has died in combat`;
+    }
+  }
+
+}
 
 // War
-class War {}
+class War {
+
+  constructor () {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+
+  addViking(vikingObject) {
+    this.vikingArmy.push(vikingObject);
+  }
+
+  addSaxon(saxonObject) {
+    this.saxonArmy.push(saxonObject);
+  }
+
+  vikingAttack() {
+    const viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+    const saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    let resultAttackViking = undefined;  
+  
+    if (this.saxonArmy.length > 0 && this.vikingArmy.length > 0) {
+      resultAttackViking = saxon.receiveDamage(viking.strength);
+      
+      if (saxon.health <= 0) {
+        this.saxonArmy.splice(saxon, 1);
+      }
+    } else {
+      resultAttackViking = this.showStatus();
+    }
+
+    return resultAttackViking;
+  }
+
+  saxonAttack() {
+    const saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    const viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+    let resultAttackSaxon = undefined
+    
+    if (this.saxonArmy.length > 0 && this.vikingArmy.length > 0) {
+      resultAttackSaxon = viking.receiveDamage(saxon.strength);
+
+      if (viking.health <= 0) {
+        this.vikingArmy.splice(viking, 1);
+      }
+    } else {
+      resultAttackSaxon = this.showStatus();
+    }
+
+    return resultAttackSaxon;
+  }
+
+  showStatus() {
+    let battleStatus = '';
+    const vikingArmyStatus = this.vikingArmy.length;
+    const saxonArmyStatus = this.saxonArmy.length;
+
+    if (vikingArmyStatus >= 1 && saxonArmyStatus >= 1) {
+        battleStatus = 'Vikings and Saxons are still in the thick of battle.';
+    } else if (this.saxonArmy.length === 0) {
+        battleStatus = 'Vikings have won the war of the century!'; 
+    } else if (this.vikingArmy.length === 0) {
+        battleStatus = 'Saxons have fought for their lives and survived another day...';
+
+    }
+
+    return battleStatus;
+  }
+
+  /**************** Extra lab *******************/
+
+  // This method is made in replace of vikingAttack() and saxonAttack()
+
+  soldierAttack(soldierType) {
+    const saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
+    const viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
+    let resultAttack = undefined;
+    let message = '';
+
+    //I know i can ask if saxon or viking is undefined but i prefer ask if the length of the array is empty.
+    if (this.saxonArmy.length > 0 && this.vikingArmy.length > 0) {
+      if (soldierType.toLowerCase() === 'saxon') {
+        resultAttack = viking.receiveDamage(saxon.strength);
+        if (viking.health <= 0) {
+          this.vikingArmy.splice(viking, 1);
```

Comment:
Como reto! (esto esta bien eeee) conoces el metodo filter de la coleccion de los Arrays? si no te diria que lo mires y me digas si no entiendes algo!

Reply:
Lo hice!!! a ver que te parece:

`soldierAttack(soldierType) {
    const saxon = this.saxonArmy[Math.floor(Math.random() * this.saxonArmy.length)];
    const viking = this.vikingArmy[Math.floor(Math.random() * this.vikingArmy.length)];
    let resultAttack = undefined;

    //I know i can ask if saxon or viking is undefined but i prefer ask if the length of the array is empty.
    if (this.saxonArmy.length > 0 && this.vikingArmy.length > 0) {
      if (soldierType.toLowerCase() === 'saxon') {
        resultAttack = viking.receiveDamage(saxon.strength);
        this.vikingArmy.filter(el => viking.health <= 0).splice(viking, 1);
      } else if (soldierType.toLowerCase() === 'viking') {
        resultAttack = saxon.receiveDamage(viking.strength);
        this.saxonArmy.filter(el => saxon.health <= 0).splice(saxon, 1);
      } else {
        throw new Error('Type of soldier wrong, please chose between: \nsaxon\nviking');
      }
    } else {
      resultAttack = this.showStatus();
    }
     
    return resultAttack;
  }
}`